### DISCUSSION TITLE
How to determine the final message of a multi packet response?

### DISCUSSION THREAD

**Discussion by ByronAP** [grblHAL/core#344] [Q&A] [answered]
If I issue a command such as $$ multiple messages are received. Am I guaranteed that only the final message will end in \r\n or is there another way to determine the end of a response?

**Comment by terjeio** [comment, marked as answer]
`ok\r\n ` terminates all multi-line messages.

**Comment by ByronAP** [comment]
I missed the ok lol thank you
